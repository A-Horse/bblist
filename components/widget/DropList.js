import React, {Component, PropTypes} from 'react';

import 'style/component/widget/drop-list.scss';

export class DropList extends Component {
  componentDidMount() {
    
  }

  componentDidUpdate() {
    
  }

  openSelect() {
    
  }

  buildClassName() {
    let className = 'drop-list';
    if (this.props.className) {
      className += ' ' + this.props.className;
    }
    return className;
  }
  
  render() {
    if (this.props.toggle) {
      return <ul className={this.buildClassName()}>
        {this.props.children}
      </ul>;
    } else {
      return React.DOM.noscript();
    }
  }
}
