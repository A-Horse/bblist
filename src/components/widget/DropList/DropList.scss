@import 'style/variable';

.drop-list {
  position: absolute;
  border: 1px solid $border-light;
  right: 0;
  top: 100%;
  padding: 0;
  margin: 0;
  background-color: $white;

  li {
    cursor: pointer;
    user-select: none;
  }
}
