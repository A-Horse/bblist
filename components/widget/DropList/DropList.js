import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DOM from 'react-dom-factories';
import R from 'ramda';
import { timeout } from 'utils/timeout';

import './DropList.scss';

export class DropList extends Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.any.isRequired,
    toggle: PropTypes.bool.isRequired
  };

  state = {
    openAfter: false,
    closeBefore: false
  };

  async componentWillReceiveProps(newProps) {
    if (newProps.toggle === true && newProps.toggle !== this.props.toggle) {
      this.setState({ openAfter: false });
      await timeout();
      this.setState({ openAfter: true });
      await timeout(200);
    }
    if (newProps.toggle === false && newProps.toggle !== this.props.toggle) {
      this.setState({ closeBefore: true });
      await timeout(200);
      this.setState({ closeBefore: false });
    }
  }

  buildClassName() {
    let className = 'drop-list';
    if (this.props.className) {
      className += ' ' + this.props.className;
    }
    if (this.state.openAfter) {
      className += ' open-after';
    }
    if (this.state.closeBefore) {
      className += ' close-before';
    }
    return className;
  }

  render() {
    if (this.props.toggle || this.state.closeBefore) {
      return (
        <ul className={this.buildClassName()} {...R.omit(['toggle', 'className'], this.props)}>
          {this.props.children}
        </ul>
      );
    } else {
      return DOM.noscript();
    }
  }
}

export default DropList;
