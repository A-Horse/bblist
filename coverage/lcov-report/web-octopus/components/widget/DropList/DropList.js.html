<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for web-octopus/components/widget/DropList/DropList.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">web-octopus/components/widget/DropList</a> DropList.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/7</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DOM from 'react-dom-factories';
&nbsp;
import './DropList.scss';
&nbsp;
export class DropList extends Component {
  componentDidMount() {}
&nbsp;
  componentDidUpdate() {}
&nbsp;
<span class="fstat-no" title="function not covered" >  op</span>enSelect() {}
&nbsp;
  buildClassName() {
    let className = 'drop-list';
    <span class="missing-if-branch" title="if path not taken" >I</span>if (this.props.className) {
<span class="cstat-no" title="statement not covered" >      className += ' ' + this.props.className;</span>
    }
    return className;
  }
&nbsp;
  render() {
    if (this.props.toggle) {
      return (
        &lt;ul className={this.buildClassName()}&gt;
          {this.props.children}
        &lt;/ul&gt;
      );
    } else {
      return DOM.noscript();
    }
  }
}
&nbsp;
export default DropList;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Sep 10 2017 22:36:46 GMT+0800 (CST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
