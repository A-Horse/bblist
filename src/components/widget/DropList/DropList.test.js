import React from 'react';
import DropList from './DropList';
import renderer from 'react-test-renderer';

test('DropList', () => {
  const tree = renderer.create(<DropList className="custom-droplist" />);
  expect(tree).toMatchSnapshot();
});
