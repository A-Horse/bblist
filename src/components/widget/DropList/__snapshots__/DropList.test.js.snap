// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropList 1`] = `<noscript />`;
